### PR TITLE
cloudapi: Add User customization

### PR DESCRIPTION
**Pull request by lavocatt**
This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] create a file in [news/unreleased](https://github.com/osbuild/osbuild-composer/tree/main/docs/news/unreleased) directory if this change should be mentioned in the release news
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

File: test/cases/api.sh [hunk 1]
```diff
@@ -407,7 +410,19 @@ function createReqFileAWS() {
           }
       }
     }
-  ]
+  ],
+    "customizations": {
+        "users":[
+            {
+                "name": "user1",
```
Comment (Gundersanne) [inline test/cases/api.sh:417]: Could we add 1 of the users to the 'wheel' group and see if it has access to sudo?

**Review by Gundersanne — COMMENTED** [review]
Looks good! Just one comment/question

File: test/cases/api.sh (hunk 1 above)
Comment (lavocatt) [reply test/cases/api.sh:417]: good idea!

Comment (lavocatt) [reply test/cases/api.sh:417]: So I've tried it:

```
sh toto@ec2-18-193-77-205.eu-central-1.compute.amazonaws.com
[toto@ip-172-31-44-249 ~]$ sudo su
[sudo] password for toto: 
[toto@ip-172-31-44-249 ~]$ groups
toto wheel
```

The issue is that sudo asks for a password, so I could instead lists the groups ?

Comment (lavocatt) [reply test/cases/api.sh:417]: So know the test looks for the presence of wheel for user1 and the absence of wheel for user2.

File: test/cases/api.sh [hunk 2]
```diff
@@ -799,6 +815,22 @@ function verifyInAWS() {
   # Verify image
   _ssh="ssh -oStrictHostKeyChecking=no -i ./keypair.pem $SSH_USER@$HOST"
   _instanceCheck "$_ssh"
+
+  # Check access to user1 and user2
+  check_groups=$(ssh -i /tmp/usertest "user1@$HOST" -t 'groups')
```
Comment (Gundersanne) [inline test/cases/api.sh:820]: Could also check straight up with `ssh user1@HOST sudo echo I have sudo`, but either is fine I suppose

**Review by Gundersanne — DISMISSED** [review]
Thanks, I'll help retry until green :)

File: test/cases/api.sh (hunk 2 above)
Comment (lavocatt) [reply test/cases/api.sh:820]: I tried to, but sudo asks for user1 password (which I don't know also).

**Review by Gundersanne — APPROVED** [review]
Thanks!

**Pull request merged**